net/httpclient.go

```diff
@@ -2,18 +2,114 @@ package net
 
 import (
 	"crypto/tls"
+	"io"
 	"net/http"
 	"net/http/httptrace"
+	"net/url"
+	"strings"
 	"time"
 
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
+	"github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/logging"
+	"github.com/zalando/skipper/secrets"
 )
 
 const (
 	defaultIdleConnTimeout = 30 * time.Second
+	defaultRefreshInterval = 5 * time.Minute
 )
 
+type Client struct {
+	client      http.Client
+	log         logging.Logger
+	sr          secrets.SecretsReader
+	secretsPath string
+	quit        chan struct{}
+}
+
+func NewClient(o Options) *Client {
+	quit := make(chan struct{})
+	if o.Log == nil {
+		o.Log = logrus.New()
+	}
+
+	tr := NewTransport(o)
+
+	sr := o.SecretsReader
+	if sr == nil {
+		if o.BearerTokenRefreshInterval == 0 {
+			o.BearerTokenRefreshInterval = defaultRefreshInterval
+		}
+		sp := secrets.NewSecretPaths(o.BearerTokenRefreshInterval)
+		if err := sp.Add(o.BearerTokenFile); err != nil {
+			o.Log.Errorf("failed to read secret: %v", err)
+		}
+		sr = sp
+	}
+
+	c := &Client{
+		client: http.Client{
+			Transport: tr,
+		},
+		log:         o.Log,
+		sr:          sr,
+		secretsPath: o.BearerTokenFile,
+		quit:        quit,
+	}
+
+	return c
+}
+
+func (c *Client) Close() {
+	close(c.quit)
+}
+
+func (c *Client) Head(url string) (*http.Response, error) {
+	req, err := http.NewRequest("HEAD", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.Do(req)
+}
+
+func (c *Client) Get(url string) (*http.Response, error) {
```

Comment:
What about content negotiation and other headers?

Reply:
my two cents: content negotiation is out of scope for this PR

Reply:
It is not only content negotiation, but the inability to communicate with many of our services without setting the correct (required) content type. @szuecs I just mentioned this because you announced to build a non-sucking net/http.Client replacement in the golang-chat room.

Reply:
@tkrop Why this is important to communicate with our services?
Maybe sent me a private message with links to show it.

Reply:
If I want to use this as a replacement of the net/http.Client ... and this is what I understood you were working on.

Reply:
If I want to use this as a replacement of the net/http.Client ... and this is what I understood you were working on.

Reply:
If I want to use this as a replacement of the net/http.Client ... and this is what I understood you were working on.

Reply:
And no, default headers on any request are usually not flexible enough to support our [RESTful API Guidelines](https://opensource.zalando.com/restful-api-guidelines/)

Reply:
We won't handle the complete API guidelines. I don't see why default headers can not provide value, if you only interact with json APIs in most cases then apply this as `Accept` default header for example.

Reply:
The reason is, that I would expect one client to support multiple concurrent request to different endpoints. So, *iff* you intend to write the wrapper generic, you should support this use case by providing a request specific header map.

Reply:
Internally discussed with @tkrop, that we do not want to support such highlevel http client features. The purpose is more to reduce the boilerplate and add common functionality: regular DNS lookups, tracing with detailed data, Bearer token injection updated from files

More highlevel clients could use this Client in order to have a solid base